draft-gpew-priv-ppm.md

```diff
@@ -704,49 +717,39 @@ struct {
   TaskID task_id;
   opaque agg_param<0..2^16-1>;  // VDAF aggregation parameter
   opaque helper_state<0..2^16>; // helper's opaque state
-  AggregateSubReq seq<1..2^24-1>;
+  AggregateSub seq<1..2^24-1>;
 } AggregateReq;
 ~~~
 
+[TODO: Split into AggregateReq and AggregateInitReq so that only the latter
+carries the aggregation parameter.]
+
 The structure contains the PPM task, an opaque, VDAF-specific aggregation
 parameter, an opaque *helper state* string, and a sequence of *sub-requests*,
 each corresponding to a unique client report. Sub-requests are structured as
 follows:
 
 ~~~
 struct {
-  Nonce nonce;                     // Equal to Report.nonce.
-  Extension extensions<4..2^16-1>; // Equal to Report.extensions.
-  EncryptedInputShare helper_share;
+  Nonce nonce;               // The report nonce
   opaque message<0..2^16-1>; // VDAF message
-} AggregateSubReq;
+} AggregateSub;
 ~~~
 
-The `nonce` and `extensions` fields have the same value as those in the
-report uploaded by the client. Similarly, the `helper_share` field is the
-`EncryptedInputShare` from the `Report` whose index in
-`Report.encrypted_input_shares` is equal to the index of `[aggregator]` in the
-task's aggregator endpoints. [OPEN ISSUE: We usually only need to send this in
-the first aggregate request. Shall we exclude it in subsequent requests
-somehow?] The remainder of the structure is dedicated to VDAF-specific request
-parameters.
+The `nonce` field has the same value as those in the leader's report share
+uploaded by the client. The remainder of the structure is dedicated to
+VDAF-specific request parameters.
 
 In order to provide replay protection, the leader preprocesses the set of
-reports it sends in the the AggregateReq as described in {{anti-replay}}. Any
+reports incident to the AggregateReq as described in {{anti-replay}}. Any
 reports filtered out by this procedure MUST be ignored.
 
 The helper handles well-formed requests as follows. (As usual, malformed
 requests are handled as described in {{errors}}.) It first looks for PPM
 parameters corresponding to `AggregateReq.task_id`. It then preprocesses the
-sub-requests as described in {{anti-replay}}. Any sub-requests filtered out by
-this procedure MUST be ignored.
-
-In addition, for any report whose nonce contains a timestamp that falls in a
-batch interval for which it has completed at least one aggregate-share request
-(see {{aggregate-share-request}}), the helper MUST send an error messsage in
-response rather than its next VDAF message. Note that this means leaders cannot
-interleave a sequence of aggregate and aggregate-share requests for a single
-batch.
+reports incident to the AggregateReq as described in {{anti-replay}}. For any
+report filtered out by this procedure, the helper MUST respond with an error
+message rather than its next VDAF message.
```

Comment:
```suggestion
sub-response rather than its next VDAF message.
```
To make it clear that filtering out an individual report does not fail the overall aggregate transaction.